sia/hostdb/weightedlist.go

```diff
@@ -17,27 +23,31 @@ type hostNode struct {
 	left  *hostNode
 	right *hostNode
 
-	hostEntry *HostEntry
+	taken     bool // Used because components.HostEntry can't be compared to nil.
+	hostEntry components.HostEntry
```

Comment:
Pointers can cause problems. I didn't see any need for it to be a pointer, so I removed the potential pitfall.